lua/lvim/core/terminal.lua

```diff
@@ -77,49 +78,69 @@ local function get_dynamic_terminal_size(direction, size)
   end
 end
 
-M.setup = function()
-  local terminal = require "toggleterm"
-  terminal.setup(lvim.builtin.terminal)
-
-  for i, exec in pairs(lvim.builtin.terminal.execs) do
-    local direction = exec[4] or lvim.builtin.terminal.direction
-
-    local opts = {
-      cmd = exec[1] or lvim.builtin.terminal.shell,
-      keymap = exec[2],
-      label = exec[3],
-      -- NOTE: unable to consistently bind id/count <= 9, see #2146
-      count = i + 100,
-      direction = direction,
-      size = function()
-        return get_dynamic_terminal_size(direction, exec[5])
-      end,
-    }
-
-    M.add_exec(opts)
-  end
-
-  if lvim.builtin.terminal.on_config_done then
-    lvim.builtin.terminal.on_config_done(terminal)
-  end
+local function keybind_toggle(keybind)
+  local Terminal = require("toggleterm.terminal").Terminal
+  local term = Terminal:new(keybind)
+  term:toggle(keybind.size, keybind.direction)
 end
 
-M.add_exec = function(opts)
-  local binary = opts.cmd:match "(%S+)"
+local function add_keybind_keymap(keybind)
+  local binary = keybind.cmd:match "(%S+)"
```

Comment:
I think calling this a `keybind` is confusing, you can go for `term_opts` instead if you just want something more descriptive than `opts`

Reply:
I renamed the functions and parameters to match the new option name. As discussed on the Discord I'm not very keen on the new name. If you have a better suggestion let me know, and I could also keep the same name and handle deprecation with a loop instead.
Either way, I can change the parameter as you suggested